### PR TITLE
[docs] Fix typos, especially in developer docs

### PR DESCRIPTION
**Pull request by stasadev**
## The Issue

Checked for typos with LanguageTool.

## How This PR Solves The Issue

Fix typos in docs and several scripts.

<a href="https://gitpod.io/#https://github.com/drud/ddev/pull/4603"></a>

### CODE REVIEW

File: docs/content/developers/release-management.md [hunk 1]
```diff
@@ -12,8 +12,8 @@
 * AUR is Arch Linux User Repository. `ddev-bin` is at `https://aur.archlinux.org/packages/ddev-bin`. The current maintainer of this is @cweagans, who can add co-maintainers.
 * The [chocolatey](https://community.chocolatey.org/packages/ddev/) package. Additional maintainers can be added at [link](https://community.chocolatey.org/packages/ddev/1.21.1/ManagePackageOwners); they could then create tokens to push it.
 * GitHub requires write access to this repository, either via permissions on the repository or on the org.
-* Apple signing and notarization requires two acces to the Localdev Foundation group on `https://developer.apple.com`. It's easy enough to add additional people.
```
Comment (rfay) [inline docs/content/developers/release-management.md:15]: I wonder how the spellchecker allowed this one.

File: docs/content/developers/release-management.md [hunk 2]
```diff
@@ -12,8 +12,8 @@
...
-* Windows signging is an awkward process that requires a dongle. When the current signing certificate expires we definitely want the simpler approach.
```
Comment (rfay) [inline docs/content/developers/release-management.md:16]: Or this one.

File: pkg/ddevapp/global_dotddev_assets/commands/host/self-upgrade [hunk 3]
```diff
@@ -12,7 +12,7 @@ case $mypath in
   "/usr/bin/ddev")
     if [[ ${OSTYPE} = "linux-gnu"* ]]; then
       if command -v apt; then echo "You seem to have an apt-installed ddev, upgrade with 'sudo apt update && sudo apt upgrade -y ddev'";
-      elif [ -f /etc/arch-release ] && command -v yay >/dev/null ; then echo "You seem to have yay-installed ddev (AUR), upgrade with 'yay -Syu ddev'";
+      elif [ -f /etc/arch-release ] && command -v yay >/dev/null ; then echo "You seem to have yay-installed ddev (AUR), upgrade with 'yay -Syu ddev-bin'";
```
Comment (rfay) [inline pkg/ddevapp/global_dotddev_assets/commands/host/self-upgrade:15]: This one is a functionality change and requires manual testing.

**Review by rfay — APPROVED** [review]
You're always amazing. We must have excluded the developer section from spell-check? @mattstein will really like this. Needs manual testing of the self-update command on arch before pull

File: pkg/ddevapp/global_dotddev_assets/commands/host/self-upgrade (hunk 3 above)
Comment (stasadev) [reply pkg/ddevapp/global_dotddev_assets/commands/host/self-upgrade:15]: Here it is:

```
# there is not such package
$ yay -Syu ddev
...
 -> Could not find all required packages:
        ddev (Target)

# search for ddev-bin package
$ yay -Ss ddev-bin
aur/ddev-bin 1.21.4-1 (+11 0.37) (Installed)
    DDEV: a local web development environment

# upgrade ddev-bin
$ yay -Syu ddev-bin
...
==> Making package: ddev-bin 1.21.4-1
...
```

**Comment by mattstein** [comment]
The spell checker [excludes the entire `developers/` directory](https://github.com/drud/ddev/blob/master/.spellcheck.yml#L28), which happened in [this commit](https://github.com/drud/ddev/commit/81ec233273208fcad39c37759abdc36ceadf8bc1) without any mention I can find. I assumed it was for a reason that’s shrouded in mystery. (Most likely working out spelling in that section will take a significant amount of effort.)

**Comment by rfay** [comment]
I think it's ok, thanks for chasing it. We can just leave it that way, and @stasadev will get us out of trouble from time to time :)

File: docs/content/developers/release-management.md [hunk 4]
```diff
@@ -40,7 +40,7 @@ The following “Repository secret” environment variables must be added to <ht
 
 ### Prerelease Tasks
 
-* Make sure [`version-history.md`](https://github.com/drud/ddev/blob/master/version-history.md) is up to date.
+* Make sure [`version-history.md`](https://github.com/drud/ddev/blob/master/version-history.md) is up-to-date.
```
Comment (mattstein) [inline docs/content/developers/release-management.md:43]: This “up to date” doesn’t need to be hyphenated:

```suggestion
* Make sure [`version-history.md`](https://github.com/drud/ddev/blob/master/version-history.md) is up to date.
```

If it was modifying a subject, then the hyphen would apply. (_Simon’s “up-to-date tie” was not as culturally relevant as he thought._ vs. _Simon’s tie, it turns out, was not up to date._)

File: docs/content/developers/release-management.md [hunk 5]
```diff
@@ -58,7 +58,7 @@ The easiest way to push Docker images is to use the GitHub Actions workflow, esp
 
 1. Push the new version of drud/ddev-php-base
 2. Update drud/ddev-webserver to use the new version of drud/ddev-php-base and push it with the proper tag
-3. Make sure the version-history.md file is up to date.
+3. Make sure the version-history.md file is up-to-date.
```
Comment (mattstein) [inline docs/content/developers/release-management.md:61]: Same as previous comment, but filenames should have backticks:

```suggestion
3. Make sure the `version-history.md` file is up to date.
```

**Review by mattstein — CHANGES_REQUESTED** [review]
Added some minor notes. Thanks for fixing these crimes against proper English!

**Pull request merged**